pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -361,6 +397,8 @@ func (r *MemberRollReconciler) reconcileObject(ctx context.Context, roll *maistr
 		return reconcile.Result{}, err
 	} else if kialiErr != nil {
 		return reconcile.Result{}, kialiErr
+	} else if prometheusErr != nil {
+		return reconcile.Result{}, prometheusErr
```

Comment:
Some time ago I added helpers to improve readability of reconciliation, so you could replace this with:
```
return common.RequeueWithError(prometheusErr)
```

I would even suggest to refactor this if-else statement entirely as you can see below. I really don't like reusing `err` variable, because it makes harder to ensure that all errors are handled properly.
```
	if err = r.updateStatus(ctx, roll); err != nil {
		return common.RequeueWithError(err)
	} else if kialiErr != nil {
		return common.RequeueWithError(kialiErr)
	} else if prometheusErr != nil {
		return common.RequeueWithError(prometheusErr)
	}
	return common.Reconciled()
```